config/server.go

```diff
@@ -34,19 +35,56 @@ type serverGRPC struct {
 	Port string
 }
 
+type serverCors struct {
+	Enabled          bool
+	AllowedOrigin    []string
+	AllowedMethods   []string
+	AllowedHeaders   []string
+	AllowCredentials bool
+	MaxAge           int
+}
+
 func serverConfigLoader() {
 	viper.SetDefault("SERVER_BATCH_DEDUP_IN_CONNECTION_ENABLED", "false")
 	Server = server{
 		DedupEnabled: util.MustGetBool("SERVER_BATCH_DEDUP_IN_CONNECTION_ENABLED"),
 	}
 }
 
+func serverCorsConfigLoader() {
+	allowedHeaders := []string{"Content-Type"}
+	allowedHeaders = setAllowedHeaders(allowedHeaders, "SERVER_WEBSOCKET_CONN_GROUP_HEADER", "")
+	allowedHeaders = setAllowedHeaders(allowedHeaders, "SERVER_WEBSOCKET_CONN_ID_HEADER", "")
+	viper.SetDefault("SERVER_CORS_ENABLED", false)
+	viper.SetDefault("SERVER_CORS_ALLOWED_ORIGIN", "*")
+	viper.SetDefault("SERVER_CORS_ALLOWED_METHODS", []string{"GET", "HEAD", "POST"})
+	viper.SetDefault("SERVER_CORS_ALLOWED_HEADERS", allowedHeaders)
+	viper.SetDefault("SERVER_CORS_ALLOW_CREDENTIALS", false)
+	viper.SetDefault("SERVER_CORS_PREFLIGHT_MAX_AGE_SECONDS", 0)
+	ServerCors = serverCors{
+		Enabled:          util.MustGetBool("SERVER_CORS_ENABLED"),
+		AllowedOrigin:    viper.GetStringSlice("SERVER_CORS_ALLOWED_ORIGIN"),
+		AllowedMethods:   viper.GetStringSlice("SERVER_CORS_ALLOWED_METHODS"),
+		AllowCredentials: util.MustGetBool("SERVER_CORS_ALLOW_CREDENTIALS"),
+		AllowedHeaders:   viper.GetStringSlice("SERVER_CORS_ALLOWED_HEADERS"),
+		MaxAge:           util.MustGetInt("SERVER_CORS_PREFLIGHT_MAX_AGE_SECONDS"),
+	}
+}
+
+func setAllowedHeaders(allowedHeaders []string, envKey string, defaultValue string) []string {
+	viper.SetDefault(envKey, defaultValue)
+	if header := viper.GetString(envKey); header != "" {
+		allowedHeaders = append(allowedHeaders, header)
+	}
+	return allowedHeaders
+}
+
 func serverWsConfigLoader() {
 	viper.SetDefault("SERVER_WEBSOCKET_PORT", "8080")
 	viper.SetDefault("SERVER_WEBSOCKET_MAX_CONN", 30000)
 	viper.SetDefault("SERVER_WEBSOCKET_READ_BUFFER_SIZE", 10240)
 	viper.SetDefault("SERVER_WEBSOCKET_WRITE_BUFFER_SIZE", 10240)
-	viper.SetDefault("SERVER_WEBSOCKET_CHECK_ORIGIN", true)
+	viper.SetDefault("SERVER_CORS_ENABLED", false)
```

Comment:
I have repeated it twice to ensure the independence from the calling order outside the function. Otherwise, it will be enforced that one has to be called after the other. WDYT

Reply:
I couldn't the calling order part

Reply:
There are two config loader function setting the default. If we do not setDefault in both the functions, it means that us setting the defaults only in one place would mean that we are dependent on the external function https://github.com/raystack/raccoon/pull/72/files#diff-9371f3d2e5b64169edcad290456eb6ecd2cf2070e675fd6568f03dfb59b5421fR28 being called after the serverWsConfigLoader

Reply:
but why is the order necessary? as long as it is being called